.github/workflows/haskell-ci.yml

```diff
@@ -172,12 +184,31 @@ jobs:
       run: cabal haddock --disable-documentation ${{ inputs.build-targets }}
       if: inputs.haddock
       working-directory: ${{ env.WORK_DIR }}
+    # See "Delete old Cabal cache" step below
+    - name: Hash the cache again
+      run: >
```

Comment:
Another nit: combining this with the `gh cache delete` as a shell `if` rather than a job-level `if` seems slightly more efficient (and I only mention this because caching is all about performance).

Reply:
Why would a shell-level `if` be more efficient than a job-level `if`?

I shuoild definitely only be hashing on a cache hit, though. Will do that.